nm-proxy/manager/manager.go

```diff
@@ -0,0 +1,366 @@
+package manager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/gravitl/netclient/nm-proxy/config"
+	"github.com/gravitl/netclient/nm-proxy/models"
+	peerpkg "github.com/gravitl/netclient/nm-proxy/peer"
+	"github.com/gravitl/netclient/nm-proxy/wg"
+	"github.com/gravitl/netmaker/logger"
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+// ProxyAction - type for proxy action
+type ProxyAction string
```

Comment:
could be shortened to Action

Reply:
I think this naming is good because this is present inside a pkg called `manager` while referencing it would make sense